### ISSUE TITLE
Inconsistent variants in the VCF outputs

### ISSUE THREAD

**Issue by AndreaGuarracino** [chfi/rs-gfa-utils#5]
Using the `gfa2vcf` functionality, some of the emitted variants are not loved by tools for working with VCF files, such as [vt](https://github.com/atks/vt).

I report an example using a dataset from the [HLA-zoo](https://github.com/ekg/HLA-zoo) repository:

```
gfautil -i ~/git/HLA-zoo/graphs/smoothxg/seqwish/mashmap/DRB1-3123.smooth.gfa gfa2vcf --refs 'gi|568815592:32578768-32589835' > test.vcf
vt normalize test.vcf -r ~/git/HLA-zoo/seqs/DRB1-3123.fa -n
```

```
...
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:48-52 - GACAG(REF) vs TACAG(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:55-57 - AGA(REF) vs TGA(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:70-76 - CGGAGGC(REF) vs GGGAGGC(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:84-89 - GTTAGG(REF) vs TTTAGG(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:90-106 - AAAAGGGGAGCACAAAA(REF) vs CAAAGGGGAGCACAAAA(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:107-109 - TTT(REF) vs GTT(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:109-119 - GGAAGATGAGG(REF) vs TGAAGATGAGG(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:121-126 - CCTGCC(REF) vs GCTGCC(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:132-134 - CGC(REF) vs TGC(FASTA)
[normalize.cpp:183 normalize] Normalization skipped due to inconsistent reference sequences
[variant_manip.cpp:75 is_ref_consistent] Variant is not consistent: gi|568815592:32578768-32589835:134-138 - GTGGG(REF) vs CTGGG(FASTA)
...
```

It seems there is a bit of noise at the first base.

**Comment by chfi** [comment]
I think I've found the issue, it's this bit (and the corresponding part in the `insertion` method: [variants.rs#L294-L299](https://github.com/chfi/rs-gfa-utils/blob/master/src/variants.rs#L294-L299)

```rust
let (prev_ref_node, _prev_ref_offset, _) = if ref_ix == 0 {
    self.ref_path[ref_ix]
} else {
    self.ref_path[ref_ix - 1]
};
```

The problem is that `detect_variants_in_sub_paths` calls `detect_variants_against_ref` on subpaths, meaning `ref_ix == 0` at the start of every subpath, and so the last base of the *current* node is used instead of the previous node.

There are many ways to fix this, but I think the best would be to redefine subpaths as ranges of indices into the paths they come from. I'll give that a shot tomorrow.

**Comment by chfi** [comment]
Fixed by 0bd5fb3082a5b401aee1cb21cdfd3971b3258628 and 517ace5ce440231dfa14b18ce57292707a4ee43f, `vt normalize` seems to be happy now.

It's a temporary hack as many related parts of [`variants.rs`](https://github.com/chfi/rs-gfa-utils/src/variants.rs) still should be rewritten.
